cmake/opentelemetry-proto.cmake

```diff
@@ -26,7 +26,7 @@
 #
 
 if(OTELCPP_PROTO_PATH)
-  if(NOT EXISTS(${OTELCPP_PROTO_PATH}/opentelemetry/proto/common/v1/common.proto))
```

Comment:
@marcalff , do you think we can have a test on this passed in env var?

Reply:
Tested locally with:

```
malff@localhost.localdomain:build-opentelemetry-cpp> pwd
/home/malff/CODE/MY_GITHUB/build-opentelemetry-cpp
```

```
ccmake \
-DOTELCPP_PROTO_PATH=/home/malff/CODE/MY_GITHUB/opentelemetry-proto \
-DWITH_OTLP=ON \
-DWITH_OTLP_HTTP=ON \
-DWITH_OTLP_GRPC=ON \
 ../opentelemetry-cpp \
```

CMake result without the fix:

```
 CMake Error at cmake/opentelemetry-proto.cmake:30 (message):
   OTELCPP_PROTO_PATH does not point to a opentelemetry-proto repository
```

CMake result with the fix:

```
 opentelemetry-proto dependency satisfied by: external path
```

So the fix is working.

OTELCPP_PROTO_PATH also shows in the ccmake UI.

@ThomsonTan 
Using environment variables will be odd, may be I don't understand what you suggest.

Reply:
@ThomsonTan  - Hope this is ok to merge. If there are any specific test scenarios, we can address them separately.